Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,259 @@
+        $return = $this->driver->consume($this->consumer->getName(), $this->queueName);
+        $this->assertNull($return);
+    }
+
+    /**
+     * Tests the consume process with a callback in return that receives the messages and ack this after
+     *
+     * @dataProvider getMessages
+     */
+    public function testConsumeWithCallbackAckingMessage(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+
+        $amqpMessage = null;
+        $callback = function(AMQPMessage $message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the consume process with a callback in return that receives the messages and nack this after
+     *
+     * @dataProvider getMessages
+     */
+    public function testConsumeWithCallbackNackingMessage(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage) {
+            $amqpMessage =  $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->nack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the function to destroy the connection and clean the variables related
+     *
+     * @dataProvider getMessages
+     * @group destroy
+     */
+    public function testDestroy(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $this->driver->declareChannel();
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->consume($this->consumer->getName(), $this->queueName);
+        $this->driver->destroy($this->consumer->getName());
+        $this->assertFalse($this->driver->isConnected());
```

Comment:
If the purpose is to test the destroy process you could simplify doing that (you don't need to have a message in the queue)
```
$this->driver->connect(); // If you want to make visible, or not a here because yo have this already in the setUp method
$this->assertTrue($this->driver->isConnected());

$this->driver->consume('consumer-name', $this->queueName); // Internally will create the channel
$this->driver->destroy($this->consumer->getName());
$this->assertFalse($this->driver->isConnected());
```

Reply:
or another option if you don't want to rely on the setUp is create the driver on the test and do a similar approach as I wrote in a comment for `testDeclareChannelWithoutConnnection()`

Reply:
Ok. Refactored.

Reply:
Test changed to not depend on setUp status, following the same way as `testConnectWithoutData`